demes/demes.py

```diff
@@ -1143,6 +1143,33 @@ def time_span(self):
         """
         return self.start_time - self.end_time
 
+    def size_at(self, time: float) -> float:
+        """
+        Get the size of the deme at a given time.
+
+        :param float time: The time at which the size should be calculated.
+        :return: The deme size.
+        :rtype: float
+        """
+        for epoch in self.epochs:
+            if epoch.start_time >= time >= epoch.end_time:
+                break
+        else:
+            raise ValueError(f"deme {self.name} doesn't exist at time {time}")
```

Comment:
The error could give the deme's start and end time as a hint, perhaps? E.g. `ValueError(f"time {time} is outside of existence interval of deme {self.name}: ({deme.end_time}, {deme.start_time})")`